controllers/vspherecluster_controller.go

```diff
@@ -230,11 +223,14 @@ func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconci
 		return reconcile.Result{}, err
 	}
 	if len(haproxyLoadbalancers.Items) > 0 {
+		conditions.MarkFalse(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
 		for _, lb := range haproxyLoadbalancers.Items {
 			if err := r.Client.Delete(ctx, lb.DeepCopy()); err != nil && !apierrors.IsNotFound(err) {
+				conditions.MarkFalse(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition, "DeletionFailed", clusterv1.ConditionSeverityWarning, "")
 				return reconcile.Result{}, err
 			}
 		}
+		conditions.MarkFalse(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
apologies, got my line numbers mixed up in original comment. I was suggesting to add this after the `if len(haproxyLoadbalancers.Items) > 0 {` block. As it stands, we requeue after a 10 second wait but may be prematurely setting the condition to Deleted.

Reply:
no problems! done!